scrabble.js

```diff
@@ -1,8 +1,88 @@
+'use strict';
+
 var Scrabble = function() {};
 
-// YOUR CODE HERE
-Scrabble.prototype.helloWorld = function() {
-  return 'hello world!';
+Scrabble.prototype.splitWord = function(word) {
+  // add some data validation here later
+  var wordString = word.toUpperCase();
+  var letterArray = Array.from(wordString);
+  return letterArray;
+}
+
+Scrabble.prototype.score = function(word) {
+  // pass in splitWord
+  var letters = this.splitWord(word);
+  var wordScore = 0;
+  // loop over each element in letters
+  for (var i = 0; i < letters.length; i++) {
+    var letter = letters[i];
+    // assign a score to letter
+    var score = this.letterScores[letter];
+    // add score to running total
+    wordScore += score;
+
+  }
+  // return score for word
+  return wordScore;
+
+}
+
+Scrabble.prototype.highestScoreFrom = function(arrayOfWords) {
+  var highScore = 0;
+  var highestScoringWord = null; // using 0 should I use null instead?
+
+  for (var i = 0; i < arrayOfWords.length; i++) {
+
+    var word = arrayOfWords[i];
+    var currentWordScore = this.score(word);
+
+    if (currentWordScore == highScore){
+      if (word.length == 7 && highestScoringWord.length != 7){
+        highestScoringWord = word;
+        // if there is a word that uses all seven tiles award that word an additional 50 pts
+        highScore = currentWordScore + 50;
+         // if there is a tie, prefer the highest scorring word made with the fewest tiles
```

Comment:
Shouldn't the `score` function already count in the 50 point 7-letter-word bonus?